lib/updatehub/updatehub_priv.h

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2018 O.S.Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ *
+ *  @brief This file contains structures representing JSON messages
+ *  exchanged with a UpdateHub
+ */
+
+#ifndef __UPDATEHUB_PRIV_H__
+#define __UPDATEHUB_PRIV_H__
+
+#define UPDATEHUB_API_HEADER \
+	"Api-Content-Type: application/vnd.updatehub-v1+json"
+
+enum updatehub_uri_path {
+	UPDATEHUB_PROBE = 0,
+	UPDATEHUB_REPORT,
+	UPDATEHUB_DOWNLOAD,
+};
+
+enum updatehub_state {
+	UPDATEHUB_STATE_DOWNLOADING = 0,
+	UPDATEHUB_STATE_DOWNLOADED,
+	UPDATEHUB_STATE_INSTALLING,
+	UPDATEHUB_STATE_INSTALLED,
+	UPDATEHUB_STATE_REBOOTING,
+	UPDATEHUB_STATE_ERROR,
+};
+
+static char *updatehub_response(enum updatehub_response response)
```

Comment:
Also, do these need to be in the header? I'm not seeing a performance reason to inline them, and regardless they aren't marked for inline anyway. This is going to cause a separate copy of this function to be built into every C file that includes (and uses) this function.

Reply:
This is used just on updatehub.c, I've separated because I wanted to turn cleaner updatehub.c.
But, if is better, I can put this on updatehub.c.